File: PHP/1079.php
<?php

class MediaPonderada{
	private $medias;

	function calcula_medias($n){
		for ($i = 0; $i < $n; $i++){
			$entry = explode(' ', readline());
			$valores = array_map('floatval', $entry);
			$this->medias[] = number_format(($valores[0]*2 + $valores[1]*3 + $valores[2]*5) / 10, 1);
		}
	}

	function imprime_medias($n){
		for ($i = 0; $i < $n; $i++){
			echo("{$this->medias[$i]}\n");
		}
	}
}

$n = (int) readline();
$media_ponderada = new MediaPonderada();
$media_ponderada->calcula_medias($n);
$media_ponderada->imprime_medias($n);
?>

File: Python/1079.py
class MediaPonderada:
    def __init__(self):
        self.__medias = []

    def calcula_medias(self, n: int):
        for i in range (0, n):
            entry = input().split()
            numbers = [float(x) for x in entry]
            self.__medias.append((numbers[0]*2 + numbers[1]*3 + numbers[2]*5) / 10)

    def imprime_medias(self, n: int):
        for i in range(0, n):
            print(f'{self.__medias[i]:.1f}')

if __name__ == '__main__':
    media_ponderada = MediaPonderada()
    n = int(input())
    media_ponderada.calcula_medias(n)
    media_ponderada.imprime_medias(n)
